### DISCUSSION TITLE
How can I specify custom pixel formats when importing?

### DISCUSSION THREAD

**Discussion by Til-Weimann** [dlemstra/Magick.NET#1169] [Help] [answered]
Hi,

I've got some 16 bit files with a custom pixel format that I want ImageMagick to read like this:

Bit 0-4: Red
Bit 5-9: Green
Bit 10-12: Blue
Bit 13-15: Alpha

So I essentially want to convert this 5-5-3-3 distribution to a conventional RGBA format (or alternatively, export each channel to its own grayscale image).

[ZIP file containing input and a possible desired output](https://www.mediafire.com/file/b3vqov9ll16ttql/image_sample.zip/file)

Is there a way to tell ImageMagick it should read the bits like that?

Thanks!

**Comment by dlemstra** [comment]
There is no support for that in ImageMagick. I would advise you to scale the values to the correct range before doing an import (`ReadPixels`).

**Comment by Til-Weimann** [comment, marked as answer]
I was beginning to think so too, but then some wizard on StackOverflow provided a solution.
https://stackoverflow.com/questions/71631519/how-to-specify-a-custom-channel-distribution-in-imagemagick
